modules/java/generator/gen_java.py

```diff
@@ -67,6 +67,12 @@ def checkFileRemap(path):
     "double[]": { "j_type" : "double[]", "jn_type" : "double[]", "jni_type" : "jdoubleArray", "suffix" : "_3D" }
 }
 
+# Defines a rule to add extra prefixes for names from specific namespaces.
+# In example, cv::fisheye::stereoRectify from namespace fisheye is wrapped as fisheye_stereoRectify
+namespaces_dict = {
+    "cv.fisheye": "fisheye"
+}
```

Comment:
Perhaps, it would be better to move this to ["misc" configuration files](https://github.com/opencv/opencv/blob/3.4.2/modules/calib3d/misc/java/gen_dict.json).